docs/language-server/protocol-language-server.md

```diff
@@ -3918,6 +3971,363 @@ standard input.
 null;
 ```
 
+## Library-Related Operations
+
+The library-related operations provide the Language Server with capabilities to
+check and modify project's edition settings, list editions published in a given
+edition, create local library projects which can be imported in the currently
+opened project and publish them.
+
+### `editions/listAvailable`
+
+Lists editions available on the system.
+
+Moreover, if `update` is set to `true`, it will download any new editions from
+the repositories and include them in the result as well.
+
+> Currently, if `update` was `true` but some downloads failed, the endpoint will
+> still return a success, just containing the editions that were already
+> available. In the future it should emit warnings using proper notification
+> channels.
+
+#### Parameters
+
+```typescript
+{
+  update: Boolean;
+}
+```
+
+#### Result
+
+```typescript
+{
+  editionNames: [String];
+}
+```
+
+### `editions/resolve`
+
+Resolves settings implied by the edition.
+
+> Currently, it only resolves the engine version, as only it is needed, but
+> other settings may be added if necesssary.
+
+#### Parameters
+
+```typescript
+{
+    editionName?: String;
+}
+```
+
+The `editionName` is optional, if it is not provided, the (unnamed) edition from
+currently open project's settings is resolved, taking into account both its
+parent edition and any locally set overrides.
```

Comment:
I'm also wondering because originally we suggested an union type here: `CurrentProject | Edition (name : String)`.

On one hand it seems cleaner and more concrete, on the other hand the optional solution is much simpler. But maybe we should go for clarity?

Reply:
Went for the union as discussed with @iamrecursion - if we need to change the API in the future, new union cases can be added without breaking compatibility.